nat-lab/conftest.py

```diff
@@ -1,25 +1,18 @@
-import pytest
+import os
 
 
-def pytest_collection_modifyitems(items):
-    # Separate test execution into layers based on platform. The biggest reason
-    # for this is to separate `linux_native` tests. If the tests are running
-    # normally, they are interleaved, e.g.:
-    # Test1-BoringTun, Test1-LinuxNative, Test2-BoringTun, Test2-LinuxNative.
-    # For whatever reason, `linux_native` tests that connect to VPN server take
-    # 30 seconds to run, if they are executed after the same BoringTun test.
-    # Given 4 tests like that, interleaving increases test execution time by 2 minutes.
+def pytest_runtest_setup(item):
+    if any(mark for mark in item.iter_markers() if mark.name == "vpn"):
+        os.system("docker exec nat-lab-vpn-01-1 bash -c 'wg setconf wg0 /wg0.conf'")
+        os.system("docker exec nat-lab-vpn-02-1 bash -c 'wg setconf wg0 /wg0.conf'")
 
-    for item in items:
-        order_index = 1
-        for mark in item.iter_markers():
-            if mark.name == "linux_native":
-                order_index = 0
-            elif mark.name == "windows":
-                order_index = 2
-            elif mark.name == "mac":
-                order_index = 3
-            elif mark.name == "long":
-                order_index = 4
-
-        item.add_marker(pytest.mark.order(order_index))
+    if any(mark for mark in item.iter_markers() if mark.name == "derp"):
+        os.system(
+            "docker exec -d nat-lab-derp-01-1 bash -c 'pkill nordderper ; nordderper'"
+        )
+        os.system(
```

Comment:
`-d` is asynchronous, so the test could start before the server has had the time to restart. Is that a concern or not?

Reply:
server starts quite fast, did not encounter any problems yet